lib/src/package.dart

```diff
@@ -325,7 +320,16 @@ class Package {
               );
       },
       isDir: (dir) => dirExists(resolve(dir)),
-    ).map(resolve).toList();
+    ).map(resolve).map((path) {
+      if (linkExists(path)) {
+        final target = Link(path).targetSync();
+        if (!fileExists(path)) {
+          throw DataException(
```

Comment:
On cycles we get a directory link here, why we don't know... maybe `isDir` returns `false`.

I think we should handle cycles explicitly. Or at-least have a comment explaining why this works.
It feels like it depends on some platform specific behavior.. or maybe there is something I'm missing.

Reply:
So, yeah, it was some weird platform-dependent behavior
unix-like: returns false for `fileExists`
windows: throws `FileSystemException` from `Directory::listSync` if directory is too deep in symlinks cycle


Wrote custom loop detection to make it reliable and error messages to be actually helpful (returns full list of visited symlinks prior to loop detection)